frontend/src/api/RestaurantApi.tsx

```diff
@@ -0,0 +1,37 @@
+import { RestaurantResponse } from '@/types';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from 'react-query';
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+export const useSearchRestaurants = (city?: string) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
+    const accessToken = await getAccessTokenSilently();
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/search/${city}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error('Failed to search restaurants');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/37#issuecomment-2405201956>#67cbd7</a><span> - 10/10/2024, 02:13 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling is too generic and doesn't provide enough information about the nature of the error, which could make debugging and error resolution difficult.
</div>
<div id="fix">
<b>Fix:</b> Enhance the error handling to include more specific error messages and potentially the HTTP status code
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -21,7 +21,7 @@
      );
 
      if (!response.ok) {
 -      throw new Error('Failed to search restaurants');
 +      throw new Error(`Failed to search restaurants: ${response.status} ${response.statusText}`);
      }
 
      return response.json();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged